### PR TITLE
sq 0.48.3 (new formula)

### PR DESCRIPTION
**Pull request by gromgit**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

### CODE REVIEW

File: Formula/s/sq.rb [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+    generate_completions_from_executable(bin/"sq", "completion")
+    (man1/"sq.1").write Utils.safe_popen_read(bin/"sq", "man")
+  end
+
+  test do
+    (testpath/"test.sql").write <<~EOS
+      create table t(a text, b integer);
+      insert into t values ('hello',1),('there',42);
+    EOS
+    system "#{Formula["sqlite"].bin}/sqlite3 test.db < test.sql"
```
Comment (SMillerDev) [inline Formula/s/sq.rb:22]:
```suggestion
    system "sqlite3 test.db < test.sql"
```

Comment (gromgit) [reply Formula/s/sq.rb]: 🤦‍♂️

**Comment by gromgit** [comment]
Weird, it fails in CI, but tests fine in a local Docker instance:
```bash
$ docker run -it homebrew/ubuntu22.04:latest 
linuxbrew@8ddd1ad9ac5c:~$ export HOMEBREW_NO_INSTALL_FROM_API=1 HOMEBREW_NO_AUTO_UPDATE=1 GH_TOKEN=<redacted>
linuxbrew@8ddd1ad9ac5c:~$ cd $(brew --repo homebrew/core)
linuxbrew@8ddd1ad9ac5c:~/.linuxbrew/Homebrew/Library/Taps/homebrew/homebrew-core$ gh pr checkout 179075
remote: Enumerating objects: 79295, done.
remote: Counting objects: 100% (22860/22860), done.
remote: Compressing objects: 100% (5/5), done.
remote: Total 79295 (delta 22855), reused 22857 (delta 22855), pack-reused 56435
Receiving objects: 100% (79295/79295), 21.60 MiB | 12.10 MiB/s, done.
Resolving deltas: 100% (63017/63017), completed with 2885 local objects.
From https://github.com/Homebrew/homebrew-core
 * [new ref]                 refs/pull/179075/head -> sq
Switched to branch 'sq'
linuxbrew@8ddd1ad9ac5c:~/.linuxbrew/Homebrew/Library/Taps/homebrew/homebrew-core$ brew install -s sq
==> Fetching dependencies for sq: go
==> Fetching go
==> Downloading https://ghcr.io/v2/homebrew/core/go/manifests/1.22.5
############################################################################################################################################################################### 100.0%
==> Downloading https://ghcr.io/v2/homebrew/core/go/blobs/sha256:f6705f2dc95a14d987f5a059b016a9e7d7ae74952bec49e1885c39cfc14d4b78
############################################################################################################################################################################### 100.0%
==> Fetching sq
==> Downloading https://github.com/neilotoole/sq/archive/refs/tags/v0.48.3.tar.gz
==> Downloading from https://codeload.github.com/neilotoole/sq/tar.gz/refs/tags/v0.48.3
############################################################################################################################################################################### 100.0%
==> Installing dependencies for sq: go
==> Installing sq dependency: go
==> Downloading https://ghcr.io/v2/homebrew/core/go/manifests/1.22.5
Already downloaded: /home/linuxbrew/.cache/Homebrew/downloads/60ee37f405e46d6dcc4c130fc1c751f34fa24a1b6b8c99abcac563ee9a24b434--go-1.22.5.bottle_manifest.json
==> Pouring go--1.22.5.x86_64_linux.bottle.tar.gz
🍺  /home/linuxbrew/.linuxbrew/Cellar/go/1.22.5: 12,869 files, 259.5MB
==> Installing sq
==> go build -ldflags=-s -w
==> Caveats
Bash completion has been installed to:
  /home/linuxbrew/.linuxbrew/etc/bash_completion.d
==> Summary
🍺  /home/linuxbrew/.linuxbrew/Cellar/sq/0.48.3: 11 files, 30.1MB, built in 56 seconds
==> Running `brew cleanup sq`...
Disable this behaviour by setting HOMEBREW_NO_INSTALL_CLEANUP.
Hide these hints with HOMEBREW_NO_ENV_HINTS (see `man brew`).
==> Caveats
==> sq
Bash completion has been installed to:
  /home/linuxbrew/.linuxbrew/etc/bash_completion.d
linuxbrew@8ddd1ad9ac5c:~/.linuxbrew/Homebrew/Library/Taps/homebrew/homebrew-core$ brew test sq
Warning: test is a developer command, so Homebrew's
developer mode has been automatically turned on.
To turn developer mode off, run:
  brew developer off

Fetching gem metadata from https://rubygems.org/.......
Fetching public_suffix 6.0.1
Fetching highline 3.0.1
Fetching erubi 1.13.0
Fetching docile 1.4.1
Installing erubi 1.13.0
Installing docile 1.4.1
Installing public_suffix 6.0.1
Fetching minitest 5.24.1
Installing highline 3.0.1
Fetching parallel 1.25.1
Installing minitest 5.24.1
Installing parallel 1.25.1
Fetching sorbet-runtime 0.5.11495
Fetching prism 0.30.0
Fetching ruby-macho 4.1.0
Installing sorbet-runtime 0.5.11495
Installing ruby-macho 4.1.0
Fetching sorbet-static 0.5.11495 (x86_64-linux)
Fetching vernier 1.1.1
Installing prism 0.30.0 with native extensions
Installing vernier 1.1.1 with native extensions
Fetching json_schemer 2.3.0
Installing json_schemer 2.3.0
Fetching rexml 3.3.2
Installing rexml 3.3.2
Fetching parser 3.3.4.0
Installing parser 3.3.4.0
Installing sorbet-static 0.5.11495 (x86_64-linux)
Fetching addressable 2.8.7
Installing addressable 2.8.7
Fetching rspec-expectations 3.13.1
Installing rspec-expectations 3.13.1
Fetching commander 5.0.0
Installing commander 5.0.0
Fetching yard-sorbet 0.9.0
Installing yard-sorbet 0.9.0
Fetching parlour 9.0.0
Installing parlour 9.0.0
Fetching rubocop 1.65.0
Installing rubocop 1.65.0
Fetching sorbet 0.5.11495
Installing sorbet 0.5.11495
Fetching sorbet-static-and-runtime 0.5.11495
Installing sorbet-static-and-runtime 0.5.11495
Fetching rubocop-performance 1.21.1
Fetching rubocop-rspec 3.0.3
Installing rubocop-performance 1.21.1
Installing rubocop-rspec 3.0.3
Fetching rubocop-sorbet 0.8.5
Installing rubocop-sorbet 0.8.5
Fetching rbi 0.1.14
Fetching spoom 1.4.1
Installing rbi 0.1.14
Installing spoom 1.4.1
Fetching tapioca 0.15.1
Installing tapioca 0.15.1
Bundle complete! 37 Gemfile dependencies, 72 gems now installed.
Bundled gems are installed into `../../../Homebrew/vendor/bundle`
Removing addressable (2.8.6)
Removing bootsnap (1.18.3)
Removing commander (4.6.0)
Removing docile (1.4.0)
Removing erubi (1.12.0)
Removing highline (2.0.3)
Removing json_schemer (2.2.1)
Removing minitest (5.23.1)
Removing msgpack (1.7.2)
Removing parallel (1.24.0)
Removing parlour (8.1.0)
Removing parser (3.3.1.0)
Removing prism (0.29.0)
Removing public_suffix (5.0.5)
Removing rbi (0.1.13)
Removing rexml (3.2.8)
Removing rspec-expectations (3.13.0)
Removing rubocop (1.64.0)
Removing rubocop-capybara (2.20.0)
Removing rubocop-factory_bot (2.25.1)
Removing rubocop-performance (1.21.0)
Removing rubocop-rspec (2.29.2)
Removing rubocop-rspec_rails (2.28.3)
Removing rubocop-sorbet (0.8.3)
Removing ruby-macho (4.0.1)
Removing sorbet (0.5.11391)
Removing sorbet-static-0.5.11391-x86_64 (linux)
Removing sorbet-static-and-runtime (0.5.11391)
Removing spoom (1.3.2)
Removing tapioca (0.14.2)
Removing vernier (1.0.1)
Removing yard-sorbet (0.8.1)
Error: sq is missing test dependencies: ncurses readline zlib sqlite
linuxbrew@8ddd1ad9ac5c:~/.linuxbrew/Homebrew/Library/Taps/homebrew/homebrew-core$ brew install sqlite
==> Downloading https://ghcr.io/v2/homebrew/core/sqlite/manifests/3.46.0
############################################################################################################################################################################### 100.0%
==> Fetching dependencies for sqlite: ncurses, readline and zlib
==> Downloading https://ghcr.io/v2/homebrew/core/ncurses/manifests/6.5
############################################################################################################################################################################### 100.0%
==> Fetching ncurses
==> Downloading https://ghcr.io/v2/homebrew/core/ncurses/blobs/sha256:7c49662d0f319baec24475d38210b2f9c754b2ec1d21a4a3ff39ce81d8605f03
############################################################################################################################################################################### 100.0%
==> Downloading https://ghcr.io/v2/homebrew/core/readline/manifests/8.2.10
############################################################################################################################################################################### 100.0%
==> Fetching readline
==> Downloading https://ghcr.io/v2/homebrew/core/readline/blobs/sha256:65181d2c0a9bd1d91ded6f7ec4a69b1110f65e875b332947e86a30aed7eab20f
############################################################################################################################################################################### 100.0%
==> Downloading https://ghcr.io/v2/homebrew/core/zlib/manifests/1.3.1
############################################################################################################################################################################### 100.0%
==> Fetching zlib
==> Downloading https://ghcr.io/v2/homebrew/core/zlib/blobs/sha256:38f2469db2ce63b70855a98e5ee27b5b5a92874e52542cbdc0b230bba1e7195f
############################################################################################################################################################################### 100.0%
==> Fetching sqlite
==> Downloading https://ghcr.io/v2/homebrew/core/sqlite/blobs/sha256:59a0e215241a611985229353273a803339372c7a075f040fe304388c0a044084
############################################################################################################################################################################### 100.0%
==> Installing dependencies for sqlite: ncurses, readline and zlib
==> Installing sqlite dependency: ncurses
==> Downloading https://ghcr.io/v2/homebrew/core/ncurses/manifests/6.5
Already downloaded: /home/linuxbrew/.cache/Homebrew/downloads/b2f8ce4e08b15832435170efd265346225ee388d5baab3d5c50e3a77c5673b5a--ncurses-6.5.bottle_manifest.json
==> Pouring ncurses--6.5.x86_64_linux.bottle.tar.gz
🍺  /home/linuxbrew/.linuxbrew/Cellar/ncurses/6.5: 4,037 files, 10.9MB
==> Installing sqlite dependency: readline
==> Downloading https://ghcr.io/v2/homebrew/core/readline/manifests/8.2.10
Already downloaded: /home/linuxbrew/.cache/Homebrew/downloads/4ddd52803319828799f1932d4c7fa8d11c667049b20a56341c0c19246a1be93b--readline-8.2.10.bottle_manifest.json
==> Pouring readline--8.2.10.x86_64_linux.bottle.tar.gz
🍺  /home/linuxbrew/.linuxbrew/Cellar/readline/8.2.10: 51 files, 2MB
==> Installing sqlite dependency: zlib
==> Downloading https://ghcr.io/v2/homebrew/core/zlib/manifests/1.3.1
Already downloaded: /home/linuxbrew/.cache/Homebrew/downloads/f68d0caf232d52f2aa586abefbbfd7e958e384d84f3967008fa83de94b5f10ae--zlib-1.3.1.bottle_manifest.json
==> Pouring zlib--1.3.1.x86_64_linux.bottle.tar.gz
🍺  /home/linuxbrew/.linuxbrew/Cellar/zlib/1.3.1: 14 files, 476KB
==> Installing sqlite
==> Pouring sqlite--3.46.0.x86_64_linux.bottle.tar.gz
🍺  /home/linuxbrew/.linuxbrew/Cellar/sqlite/3.46.0: 13 files, 6.2MB
==> Running `brew cleanup sqlite`...
Disable this behaviour by setting HOMEBREW_NO_INSTALL_CLEANUP.
Hide these hints with HOMEBREW_NO_ENV_HINTS (see `man brew`).
linuxbrew@8ddd1ad9ac5c:~/.linuxbrew/Homebrew/Library/Taps/homebrew/homebrew-core$ brew test sq    
==> Testing sq
==> sqlite3 test.db < test.sql
==> /home/linuxbrew/.linuxbrew/Cellar/sq/0.48.3/bin/sq add --active --handle @tst test.db
==> /home/linuxbrew/.linuxbrew/Cellar/sq/0.48.3/bin/sq '@tst.t | .b'
linuxbrew@8ddd1ad9ac5c:~/.linuxbrew/Homebrew/Library/Taps/homebrew/homebrew-core$ 
```

**Comment by chenrui333** [comment]
it works for me in github action linux runner as well

```
==> Testing sq
/home/linuxbrew/.linuxbrew/Homebrew/Library/Homebrew/test.rb (Formulary::FromPathLoader): loading /home/linuxbrew/.linuxbrew/Homebrew/Library/Taps/homebrew/homebrew-core/Formula/s/sq.rb
/home/linuxbrew/.linuxbrew/Homebrew/Library/Homebrew/test.rb (Formulary::FromNameLoader): loading gcc@11
==> sqlite3 test.db < test.sql
==> /home/linuxbrew/.linuxbrew/Cellar/sq/0.48.3/bin/sq add --active --handle @tst test.db
@tst  sqlite3  test.db
==> /home/linuxbrew/.linuxbrew/Cellar/sq/0.48.3/bin/sq '@tst.t | .b'
```

**Comment by chenrui333** [comment]
the go build is bit off on version thing

```
runner@fv-az1380-767:/home/linuxbrew/.linuxbrew/Homebrew/Library/Taps/homebrew/homebrew-core$ /home/linuxbrew/.linuxbrew/Cellar/sq/0.48.3/bin/sq --version
sq v0.0.0-dev    Update available: v0.48.3
```

File: Formula/s/sq.rb [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+  homepage "https://sq.io"
+  url "https://github.com/neilotoole/sq/archive/refs/tags/v0.48.3.tar.gz"
+  sha256 "46e75e2db83a6cbc98b07dbcfb23de03fc41b2b2cbc7de7aaee0425cef4fb9bb"
+  license "MIT"
+
+  depends_on "go" => :build
+  uses_from_macos "sqlite" => :test
+
+  def install
+    system "go", "build", *std_go_args(ldflags: "-s -w")
```
Comment (chenrui333) [inline Formula/s/sq.rb:12]: need to match with this, https://github.com/neilotoole/sq/blob/e86d01fdb756a212c59edd3bd6cbd5a5a1b3056d/Makefile#L6

Comment (gromgit) [reply Formula/s/sq.rb:12]: Thanks for the catch, @chenrui333! Not sure how I missed seeing the Makefile. 🤦‍♂️

File: Formula/s/sq.rb [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+  license "MIT"
+
+  depends_on "go" => :build
+
+  uses_from_macos "sqlite" => :test
+
+  def install
+    icu4c = Formula["icu4c"]
+    ENV.append "CFLAGS", "-I#{icu4c.opt_include}"
+    ENV.append "LDFLAGS", "-L#{icu4c.opt_lib}"
```
Comment (carlocab) [inline Formula/s/sq.rb:15]: Kinda weird to have this when `icu4c` isn't even a dependency?

File: Formula/s/sq.rb [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+
+  test do
+    (testpath/"test.sql").write <<~EOS
+      create table t(a text, b integer);
+      insert into t values ('hello',1),('there',42);
+    EOS
+    system "sqlite3 test.db < test.sql"
+    out1 = shell_output("#{bin}/sq add --active --handle @tst test.db")
+    assert_equal %w[@tst sqlite3 test.db], out1.strip.split(/\s+/)
+    out2 = shell_output("#{bin}/sq '@tst.t | .b' </dev/null 2>&1")
```
Comment (carlocab) [inline Formula/s/sq.rb:40]:
```suggestion
    out2 = pipe_output("#{bin}/sq '@tst.t | .b' 2>&1", "")
```
Might work here.

File: Formula/s/sq.rb (hunk 3 above)
Comment (gromgit) [reply Formula/s/sq.rb:15]: Sorry, forgot to remove those lines.

File: Formula/s/sq.rb (hunk 4 above)
Comment (gromgit) [reply Formula/s/sq.rb:40]: Nope, sorry. With `pipe_output`,  even an interactive `brew test` abends on every platform with `sq: EOF`. That `/dev/null` redirection is needed for some reason.

**Review by carlocab — APPROVED** [review]

**Review by p-linnane — APPROVED** [review]

**Pull request merged**